src/main/resources/schema/elytron-client-1_1.xsd

```diff
@@ -43,27 +48,73 @@
+            <xsd:annotation>
+                <xsd:documentation>
+                    The default SSLContext.
+                </xsd:documentation>
+            </xsd:annotation>
+            </xsd:element>
         </xsd:choice>
     </xsd:complexType>
 
     <xsd:complexType name="ssl-context-type">
+        <xsd:annotation>
+            <xsd:documentation>
+                SSLContext definition.
+            </xsd:documentation>
+        </xsd:annotation>
         <xsd:all>
             <xsd:element name="key-store-ssl-certificate" type="key-store-ref-type" minOccurs="0"/>
             <xsd:element name="trust-store" type="trust-store-ref-type" minOccurs="0"/>
             <xsd:element name="trust-manager" type="trust-manager-type" minOccurs="0"/>
-            <xsd:element name="cipher-suite" type="selector-type" minOccurs="0"/>
-            <xsd:element name="protocol" type="names-list-type" minOccurs="0"/>
-            <xsd:element name="provider-name" type="name-type" minOccurs="0"/>
+            <xsd:element name="cipher-suite" type="selector-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        Configuration to filter the enabled cipher suites. 
+                        The format of the selector is org.wildfly.security.ssl.CipherSuiteSelector.fromString(selector).
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="protocol" type="names-list-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        Define a space separated list of the protocols to be supported.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
+            <xsd:element name="provider-name" type="name-type" minOccurs="0">
+                <xsd:annotation>
+                    <xsd:documentation>
+                        Once the available providers have been identified only the provider with the name defined on this element will be used.
+                    </xsd:documentation>
+                </xsd:annotation>
+            </xsd:element>
             <xsd:element name="providers" type="providers-type" minOccurs="0"/>
             <xsd:element name="certificate-revocation-list" type="certificate-revocation-list-type" minOccurs="0"/>
         </xsd:all>
-        <xsd:attribute name="name" type="xsd:string" use="required"/>
+        <xsd:attribute name="name" type="xsd:string" use="required">
+            <xsd:annotation>
+                <xsd:documentation>
+                    SSLContext name.
+                </xsd:documentation>
+            </xsd:annotation>
+        </xsd:attribute>
     </xsd:complexType>
 
     <xsd:complexType name="ssl-context-rules-type">
+        <xsd:annotation>
+            <xsd:documentation>
+                The rules to match against the SSLContext definition.
```

Comment:
Rules used to determine which SSLContext should be used for individual resources.